### PR TITLE
refactors dehydrated carps to be less shit

### LINKED ISSUE #69950: Fun fish friends dont work :(
## Round ID:
[190562](https://scrubby.melonmesa.com/round/190562)

## Reproduction:

Buy 14 dehydrated space carp from unlink 
Pet then all inhand 
Rehydrate them via fire extinguisher
Cry as they all sit in a corner and do fucking nothing

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

this code was absolutely horrible, makes it better (still not perfect but fine enough)

## Why It's Good For The Game

this fixes https://github.com/tgstation/tgstation/issues/69950

## Changelog

:cl:
fix: makes dehydrated carps be friends again
/:cl:

### CODE REVIEW

File: code/game/objects/items/dehy_carp.dm [hunk 1]
```diff
@@ -6,69 +6,68 @@
...
 /obj/item/toy/plush/carpplushie/dehy_carp/attack_self(mob/user)
-	src.add_fingerprint(user) //Anyone can add their fingerprints to it with this
-	if(!owned)
-		to_chat(user, span_notice("You pet [src]. You swear it looks up at you."))
-		owner = user
-		owned = 1
-	else
+	if(owner)
 		return ..()
+	add_fingerprint(user)
```
Comment (JohnFulpWillard) [inline code/game/objects/items/dehy_carp.dm:16]: wtf balance change (prints are only from the first person to attack self with it)

Comment (Fikou) [reply code/game/objects/items/dehy_carp.dm:16]: the parent calls interact which calls add fingerprint

File: code/game/objects/items/dehy_carp.dm [hunk 2]
```diff
@@ -6,69 +6,68 @@
...
-			if(F == "neutral")
-				factions -= F
-		M.faction = factions
-	if (!owner || owner.faction != M.faction)
-		visible_message(span_warning("You have a bad feeling about this.")) //welcome to the hostile carp enjoy your die
-	else
-		visible_message(span_notice("The newly grown [M.name] looks up at you with friendly eyes."))
+		spawned_mob.faction = list("[REF(owner)]")
+	for(var/mob/living/viewer in viewers(5, get_turf(src)))
+		to_chat(viewer, viewer == owner ? span_warning("You have a bad feeling about this.") : span_notice("The newly grown [spawned_mob.name] looks up at you with friendly eyes."))
```
Comment (ShizCalev) [inline code/game/objects/items/dehy_carp.dm:66]: visible_message( span_notice("The newly grown [spawned_mob.name] looks up at you with friendly eyes."), vision_distance = 5, ignored_mobs = owner)
to_chat(owner, span_warning("You have a bad feeling about this."))

Comment (Fikou) [reply code/game/objects/items/dehy_carp.dm:66]: then the owner will get the message regardless of range and vision

**Pull request merged**